File: java dsa codes/stack.java
public class stack {

    class stacknode{
        int size;
        int top;
        int arr[];
    }

    public static int isempty(stacknode node){
        if(node.top==-1){
            return 1;
        }
        else{
            return 0;
        }
    }

    public static int isfull(stacknode node){
        if(node.top==node.size-1){
            return 1;
        }
        else{
            return 0;
        }
    }

    public static int push(stacknode node,int val){
        if(isfull(node)==1){
            System.out.println("stack overflow");
            return -1;
        }
        else{
            node.top++;
            node.arr[node.top]=val;
            System.out.println("the value is pushed");
            return 0;
        }
    }

    public static int pop(stacknode node){
        if(isempty(node)==1){
            System.out.println("stack underflow");
            return -1;
        }
        else{
            int val=node.arr[node.top];
            node.top--;
            System.out.println("the value is popped"+val);
            return val;
        }
    }

    public static int peek(stacknode node,int i){
        if(node.top-i+1<0){
            System.out.println("invalid position");
            return -1;
        }
        else{
            return node.arr[node.top-i+1];
        }
    }

    public static int stacktop(stacknode node){
        if(isempty(node)==1){
            System.out.println("stack is empty");
            return -1 ;
        }
        else{
            return node.arr[node.top];
        }
    }

    public static int stackbottom(stacknode node){
        if(isempty(node)==1){
            System.out.println("Stack is empty");
            return -1;
        }
        else{
            return node.arr[0];
        }
    }

    public static void main(String [] args){
        stack s=new stack();
        stacknode node=s.new stacknode();
        node.size=10;
        node.top=-1;
        node.arr=new int[node.size];

        System.out.println("stack created of succesfully");

        push(node,78);
        push(node,18);
        push(node,28);
        
        stackbottom(node);
        System.out.println("the top element is :"+stacktop(node));
        stackbottom(node);





    }
}

File: c dsa codes/stack.c
#include<stdio.h>
#include<stdlib.h>

struct stack{
    int size;
    int top;
    int * arr;
};

int isempty(struct stack * ptr){
    if(ptr->top==-1){
        return 1;
    }
    else{
        return 0;
    }
}
int isfull(struct stack * ptr){
    if(ptr->top==ptr->size-1){
        return 1;
    }
    else{
        return 0;
    }
}

int push(struct stack*ptr,int val){
    if(isfull(ptr)){
        printf("Stack overflow");
    }
    else{
        ptr->top++;
        ptr->arr[ptr->top]=val;
        printf("pushed %d to stack\n",val);
    }
}

int pop(struct stack * ptr){
    if(isempty(ptr)){
        printf("Stack underflow");
    }
    else{
        int val=ptr->arr[ptr->top];
        ptr->top--;
        return val;
    }
}

int peek(struct stack * ptr,int i){
    if((ptr->top-i-1)<0){
        printf("invalid postition");
        return -1;
    }
    else{
        return ptr->arr[ptr->top-i-1];
    }
}

int stacktop(struct stack* ptr){
    if(isempty(ptr)){
        printf("Stack is empty");
        return -1;
    }
    else{
        return ptr->arr[ptr->top];
    }
}

int stackbottom(struct stack* ptr){
    if(isempty(ptr)){
        printf("Stack is empty");
        return -1;
    }
    else{
        return ptr->arr[0];
    }
}

int main(){
    struct stack * ptr;
    ptr->size=80;
    ptr->top=-1;
    ptr->arr=(int*)malloc(ptr->size*sizeof(int));

    printf("stack created succesfully\n");
    
    ptr->arr[0]=7;
    ptr->top++;

    ptr->arr[1]=45;
    ptr->top++;

    push(ptr,78);
    printf("popped %d from stack\n",pop(ptr));

    return 0;


}